complete/src/test/java/hello/HelloWorldConfigurationTests.java

```diff
@@ -19,50 +19,53 @@
 
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.test.IntegrationTest;
-import org.springframework.boot.test.SpringApplicationConfiguration;
-import org.springframework.boot.test.TestRestTemplate;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.BDDAssertions.then;
 
 /**
  * Basic integration tests for service demo application.
  *
  * @author Dave Syer
  */
-@RunWith(SpringJUnit4ClassRunner.class)
-@SpringApplicationConfiguration(classes = HelloWorldConfiguration.class)
-@WebAppConfiguration
-@IntegrationTest({ "server.port=0", "management.port=0" })
-@DirtiesContext
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = HelloWorldConfiguration.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = {"management.port=0"})
 public class HelloWorldConfigurationTests {
 
-	@Value("${local.server.port}")
+	@LocalServerPort
 	private int port;
 
 	@Value("${local.management.port}")
 	private int mgt;
 
+	@Autowired
+	private TestRestTemplate testRestTemplate;
+
 	@Test
-	public void testGreeting() throws Exception {
+	public void should_return_200_when_sending_request_to_controller() throws Exception {
 		@SuppressWarnings("rawtypes")
```

Comment:
You don't know what's good for you... :> I will unify it then even though personally I think it's better to read it with underscores.